### ISSUE TITLE
Wide characters overlapping after gpu.copy

### ISSUE THREAD

**Issue by neumond** [MightyPirates/OpenComputers#2708]
You can get messed up wide characters with following code:
```lua
gpu.set(1, 1, "シ")
gpu.copy(1, 1, 1, 1, 1, 0)
gpu.set(3, 1, "abc")
```
![a_011](https://user-images.githubusercontent.com/3260024/34436439-0d76a432-eca7-11e7-89c7-5a3b2ee7a99d.png)

**Comment by payonel** [comment]
I tested this in both 1.7.10 and 1.12, and this is my result:
![image](https://user-images.githubusercontent.com/6590223/34438898-d3d97328-ec5e-11e7-85a2-5583c98682f3.png)

**Comment by neumond** [comment]
Wouldn't be more natural to split wide characters into subcharacters? During horizontal scrolling character parts at the edge of screen could be rendered properly instead disappearing. Again, with current behaviour you can have completely unsuccessful fills due to wrong character alignment:
```lua
gpu.fill(1, 1, 3, 3, "シ")
gpu.fill(2, 1, 2, 3, "ネ")  -- this command does nothing
```

**Comment by payonel** [comment]
I've known about problems with OC and pushing chars over wide chars -- that it didn't behave like a real terminal. And I was going to resist fixing this because this is the way we've done this for quite a long time. In fact, we have two emulators in the community and both try to mimic the weirdness of this behavior.

But as I started to test some tweaks I found that we had a pretty silly bug in our copy code that really deserved being fixed, and while I was in there I might as well make the gpu set and copy code work more like a real terminal

sooo....I have a fix now in

this is closed by 7cab77590145cf8aee1681d910ef0141b0d1e6c2
